de.dashup.application/src/main/webapp/components/dashup-grid-layout.js

```diff
@@ -0,0 +1,68 @@
+import {html} from "https://unpkg.com/lit-element@2.1.0/lit-element.js?module";
+import {classMap} from "https://unpkg.com/lit-html@1.0.0/directives/class-map.js?module";
+import {DashupComponent} from "./dashup-component.js";
+export class DashupGridLayout extends DashupComponent{
+
+    render() {
+        return html`
+            <div class="container">
+                ${this.elements.map((row) => html`
+                    <div class="row">
+                        ${row.map((element) => {
+            let size = element.layout.size;
+            let offset = element.layout.offset;
+            let classes = {col: true};
+            let sizes=["s","m","l","xl"];
+            sizes.forEach((element) => {
+                if(size){
+                    classes[element + size] = true;
+                } else {
+                    classes[element + "4"] = true;
+                }
+                if(offset){
+                    classes["offset-" + element + offset] = true;
+                }
+            });
+            return html`
+                                <div class="${classMap(classes)}">
+                                    <slot name="${this.getSlotName(element)}"></slot>
+                                </div>
+                            `;
+        })}
+                    </div>
+                `)}
+            </div>
+        `;
+    }
+
+    static get properties() {
+        return {
+            elements: {type: Array, hasChanged: () => {return false;}}
+        };
+    }
+
+    constructor() {
+        super();
+        this.elements = [];
+    }
+
+    //Called after elements templated is created -> layout for grid layout within a grid layout is already defined
+    firstUpdated() {
```

Comment:
All of the warnings are caused by the Polymer Framework and the imported LitElement library. The syntax used in the tagged template literals will be parsed to native HTML5 by the tagged html function. IntelliJ does not take this into account. Furthermore, the functions like render are unresolved because they will be added at runtime by the LitElement base class import. (If you try it out in WebStorm after importing LitElement from npm, you should not see a single warning ;) (let me know if it does))

Reply:
Are you sure I should suppress them? This would add a bunch of // noinspection comments, which in my opinion is not clean code.

Reply:
There are annotations like @suppresswarning so you can use that to suppress a single and specified warning. Use that in combination with an explaining comment and everything is fine in my opinion.

Reply:
As far as I know, the SuppressWarning annotation can only be used in java files. I would need to use //noinspection comment instead in JavaScript files, which would not be compliant to our coding conventions of removing all comments.